### ISSUE TITLE
Design UI for adding soil depth intervals at the site level

### ISSUE THREAD

**Issue by CourtneyLee333** [techmatters/terraso-product#206]

**Comment by CourtneyLee333** [comment]
Final design needs review by engineers (8/22 spec review), then probably present to stakeholders as well.

**Comment by CourtneyLee333** [comment]
Final design on hold until UT.

**Comment by CourtneyLee333** [comment]
@shrouxm @david-code Tagging you for transparency, even though this user story has not been approved for development yet.

**Comment by ltseng** [comment]
The only comment I have is that the Soil Surface and Soil Pit headings disappear after I've entered texture data, but I assume that's an oversight in the prototype rather than anything intentional?

**Comment by ltseng** [comment]
Actually, one more comment - in the prototype, the bedrock is added at 22cm, while the soil depth interval goes from 0-20, and therefore the bedrock displays in a grey bar below the 0-20 cm depth interval. How would the bedrock display if it were added at 19 cm?

**Comment by CourtneyLee333** [comment]
@ltseng Yes, the headings omission is an oversight of the prototype, thanks for catching it.

Regarding the second comment... my intention is that the placement of the bedrock indicator will the stay the same whether depth intervals were preset or chosen by the data collector. If it's important, as you say, to display the preset interval I don't know what we can do differently. Otherwise, the thought that comes to mind is to programmatically update the interval that contains the bedrock to the depth of the bedrock. Thoughts?

**Comment by ltseng** [comment]
I think that as long as the original interval is preserved, and users aren't able to create or see depth intervals lower than where bedrock has been encountered, that should be fine. Your designs are fine as-is, as long as the placement of the bedrock is consistent, i.e. outside of the depth interval.

**Comment by Hanal-code** [comment]
@CourtneyLee333 the updated heading and FAB work better.

**Comment by CourtneyLee333** [comment]
Design and notes need to be finalized for engineers.

**Comment by CourtneyLee333** [comment]
@ltseng I think the outstanding question at this point is about deleting depths, and there are several scenarios to consider. I think if we get that worked out and added to the design we can hand this off to engineers. Then, we can tackle all the other scenarios (I've left a big note for you to get that convo rolling.

**Comment by CourtneyLee333** [comment]
@ltseng Here are the scenarios where I think we need an option to delete a depth:
- Site is part of a project where the data collector defines their own depths. 
- Site is unaffiliated and data collector chooses custom depths. 

I'm thinking to add the delete option to the 3 dot menu. Sound good?

**Comment by ltseng** [comment]
Sounds good!
